lib/active_record/connection_adapters/percona_adapter.rb

```diff
@@ -119,16 +128,28 @@ def new_column(field, default, type_metadata, null, table_name, default_function
       # @param column_name [String, Symbol]
       # @param options [Hash] optional
       def add_index(table_name, column_name, options = {})
-        index_name, index_type, index_columns, index_options = add_index_options(table_name, column_name, options)
-        execute "ALTER TABLE #{quote_table_name(table_name)} ADD #{index_type} INDEX #{quote_column_name(index_name)} (#{index_columns})#{index_options}" # rubocop:disable Metrics/LineLength
+        if ActiveRecord::VERSION::STRING >= '6.1'
+          index, algorithm, if_not_exists = add_index_options(table_name, column_name, options)
+          create_index = CreateIndexDefinition.new(index, algorithm, if_not_exists)
+          execute schema_creation.accept(create_index)
+        else
+          index_name, index_type, index_columns, index_options = add_index_options(table_name, column_name, options)
+          execute "ALTER TABLE #{quote_table_name(table_name)} ADD #{index_type} INDEX #{quote_column_name(index_name)} (#{index_columns})#{index_options}" # rubocop:disable Metrics/LineLength
+        end
```

Comment:
This generates/executes different sql tested [here](https://github.com/departurerb/departure/pull/58/files#diff-c150ab33c3d9469f6de8620566924b97e6b089ee95dff7743c733b5fdbebedf7R125). Should it keep the sql same? Not sure if `ALTER TABLE` is treated differently.

```suggestion
        if ActiveRecord::VERSION::STRING >= '6.1'
          index, _, _ = add_index_options(table_name, column_name, **options)
          index_name = index.name
          index_type = index.type
          index_columns = index.columns.first.to_s
          index_options = index.column_options.compact_blank.presence
        else
          index_name, index_type, index_columns, index_options = add_index_options(table_name, column_name, options)
        end
        execute "ALTER TABLE #{quote_table_name(table_name)} ADD #{index_type} INDEX #{quote_column_name(index_name)} (#{index_columns})#{index_options}" # rubocop:disable Metrics/LineLength
      end
```